src/Symfony/Component/Security/Guard/Firewall/GuardAuthenticationListener.php

```diff
@@ -92,20 +94,45 @@ public function handle(GetResponseEvent $event)
         }
     }
 
-    private function executeGuardAuthenticator($uniqueGuardKey, GuardAuthenticatorInterface $guardAuthenticator, GetResponseEvent $event)
+    private function executeGuardAuthenticator($uniqueGuardKey, AuthenticatorInterface $guardAuthenticator, GetResponseEvent $event)
     {
         $request = $event->getRequest();
         try {
             if (null !== $this->logger) {
                 $this->logger->debug('Calling getCredentials() on guard configurator.', array('firewall_key' => $this->providerKey, 'authenticator' => get_class($guardAuthenticator)));
             }
 
+            // abort the execution of the authenticator if it doesn't support the request.
+            if ($guardAuthenticator instanceof GuardAuthenticatorInterface) {
+                // deprecated since version 3.4, to be removed in 4.0
+                @trigger_error(sprintf('The %s interface is deprecated since version 3.4 and will be removed in 4.0. Implement %s instead.', GuardAuthenticatorInterface::class, AuthenticatorInterface::class), E_USER_DEPRECATED);
```

Comment:
this runtime deprecation should be removed: it has already been trigger by DebugClassLoader